src/Database/Relations/AttachOneOrMany.php

```diff
@@ -166,6 +166,21 @@ public function add(Model $model, $sessionKey = null)
         }
 
         if ($sessionKey === null) {
+            /**
+             * @event model.relation.beforeAdd
+             * Called before adding a relation to the model (for AttachOneOrMany, HasOneOrMany & MorphOneOrMany relations)
+             *
+             * Example usage:
+             *
+             *     $model->bindEvent('model.relation.beforeAdd', function (string $relationName, \October\Rain\Database\Model $relatedModel) use (\October\Rain\Database\Model $model) {
+             *         if ($relationName === 'dummyRelation') {
+             *             throw new \Exception("Invalid relation!");
+             *         }
+             *     });
+             *
+             */
+            $this->parent->fireEvent('model.relation.beforeAdd', [$this->relationName, $this->related]);
```

Comment:
Last observation: AttachOneOrMany is special case of HasOneOrMany, so `$this->related` could be replaced with `$model` too. After that it will be done for me.

Reply:
done @Rike-cz !

Reply:
@mjauvin I think not `$this->model`, but `$model` only.

Reply:
oops, my bad... fixed.